Feat/hard delete

hafizjavaid opened the pull request.

MalcolmJohnston committed:
Added interfaces for hard delete and comments for changes to Create method so deleted SoftDelete rows can be re-created.

hafizjavaid committed:
feat: adding hard delete feature and tests

MalcolmJohnston commented on Entatea/Entatea/BaseDataContext.cs:
```diff
@@ -65,6 +65,10 @@ public async Task<T> Create<T>(T entity) where T : class
             // set value on insert on any soft delete properties
             if (classMap.IsSoftDelete)
             {
+                // TODO: if this is also an assigned key, then HardDelete this record first
```
Check out this TODO :)

hafizjavaid committed:
fix: missing todo ( if assigned keys, then hard delete first )

MalcolmJohnston committed:
Updated hard delete tests.
Updated InMemoryDataContext so that it takes into account SoftDelete attribute when deleting.
All tests passing

MalcolmJohnston committed:
Rolled SqlServer package number.
